pkg/rpcserver/rpcserver.go

```diff
@@ -389,7 +389,7 @@ func (serv *Server) connectionLoop(runner *Runner) error {
 			}
 		}
 		for len(runner.requests)-len(runner.executing) < 2*serv.cfg.Procs {
-			req := serv.execSource.Next()
+			req, _ := serv.execSource.Next()
```

Comment:
It should shutdown and  exit on stop?

Reply:
Updated the code to wait for all requests to complete and exit.